hubspot/search.go

```diff
@@ -0,0 +1,61 @@
+package hubspot
+
+import (
+	"context"
+	"time"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/spyzhov/ajson"
+)
+
+// search uses the POST /search endpoint to filter object records and return the result.
+// This is used when Since is set. Otherwise, the Read endpoint is used.
+// This endpoint paginates using paging.next.after which is to be used as an offset.
+// Read more @ https://developers.hubspot.com/docs/api/crm/search
+func (c *Connector) search(ctx context.Context, config common.ReadParams) (*common.ReadResult, error) {
+	var (
+		data *ajson.Node
+		err  error
+	)
+
+	// If filtering is not required, then we have to use the read endpoint.
+	if !requiresFiltering(config) {
+		return c.Read(ctx, config)
+	}
```

Comment:
This seems unnecessary since you already do the same check in `Read`

Reply:
This is to avoid any accidental calls directly to `search`. I agree it's unnecessary but should be a good guard clause to have in case.